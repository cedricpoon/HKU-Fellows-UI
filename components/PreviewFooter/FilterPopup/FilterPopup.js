import React, { Component } from 'react';
import { Button, Icon, Text } from 'native-base';

import PopupMenu from '../PopupMenu/PopupMenu';

class FilterPopup extends Component {

  render() {
    const { ...restProps } = this.props;

    return(
      <PopupMenu { ...restProps } /* Proptypes handling on <PopupMenu /> */ >
        <Button full transparent info iconRight>
          <Text>From Moodle</Text>
          <Icon name="at"></Icon>
        </Button>
        <Button full transparent success iconRight>
          <Text>Latest Post</Text>
          <Icon name="time"></Icon>
        </Button>
        <Button full transparent success iconRight>
          <Text>Most Replied</Text>
          <Icon name="undo"></Icon>
        </Button>
        <Button full transparent success iconRight>
          <Text>Popularity</Text>
          <Icon name="people"></Icon>
        </Button>
      </PopupMenu>
    );
  }
}

export default FilterPopup;
