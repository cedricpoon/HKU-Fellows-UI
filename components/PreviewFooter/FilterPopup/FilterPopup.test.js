import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';

import FilterPopup from './FilterPopup';

describe('Testing FilterPopup Component', () => {
  const requiredProps = {
    position: {x: 0, y: 0},
    toggle: () => {},
    onFilterThunk: () => {}
  };

  it('renders as expected with PopupMenu required props', () => {
    const wrapper = shallow(
      <FilterPopup {...requiredProps}>
        <View></View>
      </FilterPopup>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has onRef', () => {
    const wrapper = shallow(
      <FilterPopup {...requiredProps} onRef={() => {}}>
        <View></View>
      </FilterPopup>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
