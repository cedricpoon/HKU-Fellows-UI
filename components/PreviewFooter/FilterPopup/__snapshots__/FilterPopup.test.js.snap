// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FilterPopup Component renders as expected with PopupMenu required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterPopup
    position={0}
    toggle={[Function]}
  >
    <View />
  </FilterPopup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Button)
          full={true}
          iconRight={true}
          info={true}
          transparent={true}
        >
          <Styled(Text)>
            From Moodle
          </Styled(Text)>
          <Styled(Icon)
            name="link"
            type="MaterialIcons"
          />
        </Styled(Button)>,
        <Styled(Button)
          full={true}
          iconRight={true}
          success={true}
          transparent={true}
        >
          <Styled(Text)>
            Latest Post
          </Styled(Text)>
          <Styled(Icon)
            name="clock-fast"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>,
        <Styled(Button)
          full={true}
          iconRight={true}
          success={true}
          transparent={true}
        >
          <Styled(Text)>
            Most Replied
          </Styled(Text)>
          <Styled(Icon)
            name="undo"
          />
        </Styled(Button)>,
        <Styled(Button)
          full={true}
          iconRight={true}
          success={true}
          transparent={true}
        >
          <Styled(Text)>
            Popularity
          </Styled(Text)>
          <Styled(Icon)
            name="people"
            type="MaterialIcons"
          />
        </Styled(Button)>,
      ],
      "position": 0,
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Text)>
              From Moodle
            </Styled(Text)>,
            <Styled(Icon)
              name="link"
              type="MaterialIcons"
            />,
          ],
          "full": true,
          "iconRight": true,
          "info": true,
          "transparent": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "From Moodle",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "From Moodle",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "link",
              "type": "MaterialIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Text)>
              Latest Post
            </Styled(Text)>,
            <Styled(Icon)
              name="clock-fast"
              type="MaterialCommunityIcons"
            />,
          ],
          "full": true,
          "iconRight": true,
          "success": true,
          "transparent": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Latest Post",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Latest Post",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "clock-fast",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Text)>
              Most Replied
            </Styled(Text)>,
            <Styled(Icon)
              name="undo"
            />,
          ],
          "full": true,
          "iconRight": true,
          "success": true,
          "transparent": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Most Replied",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Most Replied",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "undo",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Text)>
              Popularity
            </Styled(Text)>,
            <Styled(Icon)
              name="people"
              type="MaterialIcons"
            />,
          ],
          "full": true,
          "iconRight": true,
          "success": true,
          "transparent": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Popularity",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Popularity",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "people",
              "type": "MaterialIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Button)
            full={true}
            iconRight={true}
            info={true}
            transparent={true}
          >
            <Styled(Text)>
              From Moodle
            </Styled(Text)>
            <Styled(Icon)
              name="link"
              type="MaterialIcons"
            />
          </Styled(Button)>,
          <Styled(Button)
            full={true}
            iconRight={true}
            success={true}
            transparent={true}
          >
            <Styled(Text)>
              Latest Post
            </Styled(Text)>
            <Styled(Icon)
              name="clock-fast"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
          <Styled(Button)
            full={true}
            iconRight={true}
            success={true}
            transparent={true}
          >
            <Styled(Text)>
              Most Replied
            </Styled(Text)>
            <Styled(Icon)
              name="undo"
            />
          </Styled(Button)>,
          <Styled(Button)
            full={true}
            iconRight={true}
            success={true}
            transparent={true}
          >
            <Styled(Text)>
              Popularity
            </Styled(Text)>
            <Styled(Icon)
              name="people"
              type="MaterialIcons"
            />
          </Styled(Button)>,
        ],
        "position": 0,
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Text)>
                From Moodle
              </Styled(Text)>,
              <Styled(Icon)
                name="link"
                type="MaterialIcons"
              />,
            ],
            "full": true,
            "iconRight": true,
            "info": true,
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "From Moodle",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "From Moodle",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "link",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Text)>
                Latest Post
              </Styled(Text)>,
              <Styled(Icon)
                name="clock-fast"
                type="MaterialCommunityIcons"
              />,
            ],
            "full": true,
            "iconRight": true,
            "success": true,
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Latest Post",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Latest Post",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "clock-fast",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Text)>
                Most Replied
              </Styled(Text)>,
              <Styled(Icon)
                name="undo"
              />,
            ],
            "full": true,
            "iconRight": true,
            "success": true,
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Most Replied",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Most Replied",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "undo",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Text)>
                Popularity
              </Styled(Text)>,
              <Styled(Icon)
                name="people"
                type="MaterialIcons"
              />,
            ],
            "full": true,
            "iconRight": true,
            "success": true,
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Popularity",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Popularity",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "people",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
